util/pkg/vfs/s3context.go

```diff
@@ -156,26 +164,61 @@ func (s *S3Context) getRegionForBucket(bucket string) (string, error) {
 	}
 
 	if err != nil {
-		return "", err
+		return bucketDetails, err
 	}
 
 	if response.LocationConstraint == nil {
 		// US Classic does not return a region
-		region = "us-east-1"
+		bucketDetails.region = "us-east-1"
+		bucketDetails.defaultEncryption = s.checkDefaultEncryption(bucketDetails.region, bucket)
 	} else {
-		region = *response.LocationConstraint
+		bucketDetails.region = *response.LocationConstraint
 		// Another special case: "EU" can mean eu-west-1
-		if region == "EU" {
-			region = "eu-west-1"
+		if bucketDetails.region == "EU" {
+			bucketDetails.region = "eu-west-1"
 		}
+		bucketDetails.defaultEncryption = s.checkDefaultEncryption(bucketDetails.region, bucket)
 	}
-	glog.V(2).Infof("Found bucket %q in region %q", bucket, region)
+	glog.V(2).Infof("Found bucket %q in region %q with default encryption set to %t", bucket, bucketDetails.region, bucketDetails.defaultEncryption)
 
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
-	s.bucketLocations[bucket] = region
+	s.bucketDetails[bucket] = bucketDetails
+
+	return bucketDetails, nil
+}
+
+func (s *S3Context) checkDefaultEncryption(region string, bucket string) bool {
+	client, err := s.getClient(region)
+	if err != nil {
+		glog.Warningf("Unable to read bucket encryption policy in region %q: will encrypt using AES256", region)
+		return false
+	}
+
+	glog.V(4).Infof("Checking default bucket encryption %q", bucket)
+
+	request := &s3.GetBucketEncryptionInput{}
+	request.Bucket = aws.String(bucket)
+
+	glog.V(8).Infof("Calling S3 GetBucketEncryption Bucket=%q", bucket)
+
+	result, err := client.GetBucketEncryption(request)
+	if err != nil {
+		// the following cases might lead to the operation failing:
+		// 1. A deny policy on s3:GetEncryptionConfiguration
+		// 2. No default encryption policy set
+		glog.Warningf("Unable to read bucket encryption policy: will encrypt using AES256")
```

Comment:
As a style thing, IMO it's often clearer to have the function return the value & error, and then have the caller decide whether to ignore the error.